partiql-eval/src/main/kotlin/org/partiql/eval/internal/exclude/CompiledExcludeItem.kt

```diff
@@ -0,0 +1,229 @@
+package org.partiql.eval.internal.exclude
+
+/**
+ * Internal representation of an `EXCLUDE` expr step.
+ */
+internal sealed class ExcludeStep {
+    internal data class StructField(val attr: String, val caseSensitivity: ExcludeFieldCase) : ExcludeStep()
+    internal object StructWildcard : ExcludeStep()
+    internal data class CollIndex(val index: Int) : ExcludeStep()
+    internal object CollWildcard : ExcludeStep()
+}
+
+/**
+ * Internal representation of an `EXCLUDE` struct attribute case-sensitivity.
+ */
+internal enum class ExcludeFieldCase {
+    INSENSITIVE, SENSITIVE
+}
+
+/**
+ * Represents all the compiled `EXCLUDE` paths that start with the same [CompiledExcludeItem.root]. This variant of
+ * [ExcludeNode] represents the top-level root node of the exclude tree.
+ *
+ * Notably, redundant paths (i.e. exclude paths that exclude values already excluded by other paths) will be removed.
```

Comment:
I believe the planner should be responsible for removing redundant paths.

Reply:
From offline discussion before the holiday break, I moved the redundant path elimination to the `PlanTyper` so the planner and not evaluator is responsible. The plan representation for `EXCLUDE` was also changed to better accommodate for the tree-like representation used by this elimination step and evaluation.